### PR TITLE
SPCTR-1963 New Block: Modal - Initial base logic

### PR DESCRIPTION
**Pull request by sushmak02**
### Description
This PR includes base logic for the modal popup. Like on click of a text a popup should open inside an editor. The popup inside an editor should have block inserter support.

### Screenshots
https://share.getcloudapp.com/Jru85y9b

### Types of changes

New

### How has this been tested?
- Drag drop modal block
- On click of text a popup should open

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/modal/render.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		setAttributes,
+	} = props.parentProps;
+
+	const {
+		block_id,
+		triggerText
+	} = attributes
+
+	return (
+		<React.Fragment>
```
Comment (RajkiranBagal) [inline src/blocks/modal/render.js:22]: We can just use `<>`

File: src/blocks/modal/render.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			<div className={ classnames(
+				`uagb-block-${ block_id }`,
+				'wp-block-uagb-modal'
```
Comment (RajkiranBagal) [inline src/blocks/modal/render.js:25]: This class is added by Gutenberg itself. Why do we need it?

File: src/blocks/modal/save.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	const {
+		block_id,
+		triggerText
+	} = props.attributes;
+
+	return (
+		<div
+			className={ classnames(
+				`uagb-block-${ block_id }`,
+				'wp-block-modal'
```
Comment (RajkiranBagal) [inline src/blocks/modal/save.js:15]: Why do we need this class? Also It's different from Render.

File: src/blocks/modal/save.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+			) }
+		>
+			<RichText.Content
+				value={ triggerText }
+				tagName="p"
+				className="uagb-modal-text uagb-modal-trigger"
+			/>
+			<div className="uagb-modal-popup">
+				<div className="uagb-modal-popup-wrap">
```
Comment (RajkiranBagal) [inline src/blocks/modal/save.js:24]: Avoid using multiple div's nesting wherever possible for optimized markup

File: src/blocks/modal/style.scss [hunk 5]
```diff
@@ -0,0 +1,105 @@
+body.hide-scroll {
+	overflow: hidden;
+}
+
+.uagb-modal-popup {
+	display: none;
```
Comment (RajkiranBagal) [inline src/blocks/modal/style.scss:6]: Make good use of SCSS. Write nested CSS. It will decrease CSS file size.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/modal/render.js (hunk 1 above)
Comment (sushmak02) [reply src/blocks/modal/render.js]: Done.

File: src/blocks/modal/render.js (hunk 2 above)
Comment (sushmak02) [reply src/blocks/modal/render.js:25]: Removed

File: src/blocks/modal/save.js (hunk 3 above)
Comment (sushmak02) [reply src/blocks/modal/save.js:15]: Removed

File: src/blocks/modal/save.js (hunk 4 above)
Comment (sushmak02) [reply src/blocks/modal/save.js:24]: Yes, for modal overlay and popup content this HTML is needful.

File: src/blocks/modal/style.scss (hunk 5 above)
Comment (sushmak02) [reply src/blocks/modal/style.scss:6]: Done

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**